### DISCUSSION TITLE
github.dev branch desynchronization (a bug, I guess)

### DISCUSSION THREAD

**Discussion by l-Fingon-l** [community/community#5629] [Codespaces]
# Hi.
 I have just noticed a weird thing in github.dev editor. The issue is as follows:

- I had a project with 1 main branch. Then:

  1. I had made an **additional branch**
  2. committed some changes
  3. merged it with the main branch via _merge request_
  4. deleted the **additional branch**
- Now the weird thing. When I go to the project page and press "." it opens VS code in browser (github.dev) and everything is fine. And the story of a changed file **ancestry.py** is correct as well. 
![image](https://user-images.githubusercontent.com/70348701/133167451-aa3a26d6-5451-4a0a-8d8c-08ccfcae8d00.png)
- But now if I go to the **ancestry.py** file itself and open it in Github.com and **after** that press "." the issue appears. It opens VS code as before, but what I see now is a wrong file history. The changes that were already committed and merged seem to be just pending now. 
![image](https://user-images.githubusercontent.com/70348701/133167563-a6627fa9-38c1-4603-b821-117b939e8333.png)
![image](https://user-images.githubusercontent.com/70348701/133167706-c4d60452-591a-4c9d-bdd5-02cd6d3be726.png)
- I have just one main branch left.
![image](https://user-images.githubusercontent.com/70348701/133167843-9691243a-d9fb-4c1f-a239-028c447fc6f4.png)
---
It feels like a bug to me, but I'm not really sure. Any ideas would be hugely appreciated!

**Comment by bamurtaugh** [comment]
Thanks for reporting @l-Fingon-l! @joyceerhl thoughts? Would it be helpful to track this via an issue in https://github.com/microsoft/vscode-remote-repositories-github/issues?

**Reply by bamurtaugh** [reply]
Filed https://github.com/microsoft/vscode-remote-repositories-github/issues/77 to track this in the appropriate repo.

**Reply by joyceerhl** [reply]
I'm unable to reproduce this behavior based on the description. Here's what I did
1. Open repo
2. Create branch foo
3. Commit a change to a file
4. Create PR
5. Merge PR into main
6. Go to github.com 
7. Open the file
8. Hit . to open github.dev
9. The timeline reflects the changes I made to the file. I do not see the 'Uncommitted changes'.

@l-Fingon-l Are those steps also what you did to create the bug? Can you provide a screen recording  (e.g. using gifcap.dev)? Can you also use the GitHub Repositories: Export Debug Information command to generate debug logs if you encounter this behavior again please?

**Reply by l-Fingon-l** [reply]
@joyceerhl no problem! Lemme try to reproduce

**Reply by l-Fingon-l** [reply]
> use the GitHub Repositories: Export Debug Information command to generate debug logs

Could you elaborate on how may I do this, please?

**Reply by bamurtaugh** [reply]
@l-Fingon-l `GitHub Repositories: Export Debug Information` is a command you can execute from the command palette (`F1`) in github.dev / VS Code. 

It should open up a tab with debug information within the editor that you could then share.

If this does not seem to work for you, or you do not see the command, please let us know!
